test/compress/pure_funcs.js

```diff
@@ -293,3 +293,16 @@ unary: {
         bar();
     }
 }
+
+issue_2629: {
+    options = {
+        side_effects: true,
+    }
+    input: {
+        /*@__PURE__*/ f1();
+        (/*@__PURE__*/ f2)();
+        /*@__PURE__*/ (f3());
+        (/*@__PURE__*/ f4());
```

Comment:
Other test cases should behave like v3.2.2 (greedy AST_Call.expression):
```js
/*@__PURE__*/f5(1)(2)(3);
/*@__PURE__*/f6.x(1).y(2).z(3);
```
These should work same as above:
```js
( /*@__PURE__*/f7(1)(2)(3) );
( /*@__PURE__*/f8.x(1).y(2).z(3) );
```
Also please add the sequence and array tests from https://github.com/mishoo/UglifyJS2/issues/2629#issuecomment-353139027 to make sure no regressions from v3.2.2.

Reply:
Done in https://github.com/mishoo/UglifyJS2/pull/2633/commits/fd267ad1a55e586d35497ba72539c2437036767b